packages/vm/src/modules/in_memory_cache.rs

```diff
@@ -21,20 +22,18 @@ impl InMemoryCache {
     }
 
     pub fn store(&mut self, checksum: &Checksum, module: Module) -> VmResult<()> {
-        let serialized_artifact = module.serialize()?;
-        self.artifacts.put(*checksum, serialized_artifact);
+        let artifact = Arc::clone(module.artifact());
```

Comment:
This does look better/faster than storing serialized bytes.

Still wondering why you didn't just store the `Module` directly. What was the issue with that?

Reply:
> Still wondering why you didn't just store the `Module` directly. What was the issue with that?

The module contains stuff we don't want to reuse: the `Store` with memory limits and stuff. That we really want to cache is artifacts.

When Wasmer offers you Module serialization/deserialization, it really gives you [artifact serialization](https://github.com/wasmerio/wasmer/blob/22c7bc8e19de54090f3f0a2197ad62c4a5b72c95/lib/api/src/module.rs#L179-L181)/[artifact deserialization](https://github.com/wasmerio/wasmer/blob/22c7bc8e19de54090f3f0a2197ad62c4a5b72c95/lib/api/src/module.rs#L197-L199).

The idea of serializing modules it just too simplified for our advanced use case.

Reply:
Nice link there. I see how modules are just a wrapper around `Arc<Artifact>` (cheap) and a `Store` holding the runtime info. Makes sense now